docs/instrumentation/python-fastapi-uvicorn.md

```diff
@@ -0,0 +1,99 @@
+---
+id: python-fastapi-uvicorn
+title: "Python FastAPI Uvicorn"
+slug: /instrumentation/python-fastapi-uvicorn
+---
+
+## Prerequisites
+
+Python 3
+
+## Installation
+
+1. Install dependencies:
+
+   ```shell
+   pip install opentelemetry-distro opentelemetry-exporter-otlp-proto-http opentelemetry-instrumentation-fastapi
+   opentelemetry-bootstrap -a install
+   ```
+
+2. Create a file `tracing.py` in your project directory, with the following content:
+
+   ```python title="tracing.py"
+    import os
+    from opentelemetry import trace
+    from opentelemetry.semconv.resource import ResourceAttributes
+    from opentelemetry.sdk.trace import TracerProvider, Resource
+    from opentelemetry.sdk.trace.export import (
+        BatchSpanProcessor,
+        ConsoleSpanExporter,
+        SimpleSpanProcessor,
+    )
+    from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+    from socket import gethostname
+
+
+    def init_tracing():
+        provider = TracerProvider(resource=Resource({
+           ResourceAttributes.SERVICE_NAME: os.environ['OTEL_SERVICE_NAME'],
+           ResourceAttributes.HOST_NAME: gethostname() or 'UNSET',
+        }))        
+        if os.getenv('OTEL_LOG_LEVEL', '') == 'debug':
+            processor = SimpleSpanProcessor(ConsoleSpanExporter())
+        else:
+            processor = BatchSpanProcessor(OTLPSpanExporter())
+        provider.add_span_processor(processor)
+        trace.set_tracer_provider(provider)
+
+   ```
+
+3. Add the highlighted lines below to your project's main file:
+
+   ```python title="main.py"
+   from fastapi import FastAPI
+   from tracing import init_tracing
```

Comment:
This line also needs to be highlighted. So, replace highlight-next-line below with highlight start and highlight end